#!/usr/bin/env python
PACKAGE = "fields2cover_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("op_width", double_t, 0, "Operation width",    3, 0.1,   50)
gen.add("turn_radius", double_t, 0, "Turning radius",    2.0, 0.1,   50)
gen.add("headland_width", double_t, 0, "Headland width",    6, 0,   50)
gen.add("swath_angle", double_t, 0, "Angle cropped",    .01, 0,   2*3.1416)
gen.add("automatic_angle", bool_t, 0, "Search best angle", False)
gen.add("sg_objective", int_t, 0, "SG Objective", 0, 0, 2)
gen.add("route_type", int_t, 0, "Routing alg type", 0, 0, 3)
gen.add("turn_type", int_t, 0, "Turn alg type", 0, 0, 3)

exit(gen.generate(PACKAGE, "fields2cover_ros", "F2C"))

